recipe/meta.yaml

```diff
@@ -186,10 +192,15 @@ outputs:
         - {{ pin_compatible("libcxx", max_pin=None) }}  # [osx]
     test:
       commands:
-        - test -f "$PREFIX/lib/libclang.so"                     # [linux]
-        - test -f "$PREFIX/lib/libclang.so.{{ major_version }}"   # [linux]
-        - test -f "$PREFIX/lib/libclang.dylib"                  # [osx]
-        - if not exist %LIBRARY_BIN%\\libclang.dll exit 1       # [win]
+        - test -f "$PREFIX/lib/libclang.so"                           # [linux]
+        - test -f "$PREFIX/lib/libclang.so.{{ minor_aware_ext }}"     # [linux]
+        {% if minor_aware_ext != major_version %}
+        - test ! -f "$PREFIX/lib/libclang.so.{{ major_version }}"     # [linux]
+        - test ! -f "$PREFIX/lib/libclang.{{ major_version }}.dylib"  # [osx]
+        {% endif %}
+        - test -f "$PREFIX/lib/libclang.dylib"                        # [osx]
+        - test -f "$PREFIX/lib/libclang.{{ minor_aware_ext }}.dylib"  # [osx]
+        - if not exist %LIBRARY_BIN%\\libclang.dll exit 1             # [win]
 
   - name: clang-{{ major_version }}
```

Comment:
Maybe worth a comment that this package hasn't changed name - I haven't been able to follow which packages should reflect the minor version and which shouldn't; just double-checking.